sap_business_document_processing/document_classification_client/dc_api_client.py

```diff
@@ -0,0 +1,472 @@
+        classification_job = response.json()
+        result = self._poll_for_url(DOCUMENT_RESULT_ENDPOINT(modelName=model_name,
+                                                             modelVersion=model_version,
+                                                             id=classification_job[API_DOCUMENT_ID_FIELD]),
+                                    wait_status=409).json()
+        return result
+
+    def _classify_document_wrap_errors(self,
+                                       document_path,
+                                       model_name,
+                                       model_version,
+                                       reference_id=None,
+                                       mime_type=None):
+        result = function_wrap_errors(self.classify_document, document_path, model_name, model_version,
+                                      reference_id, mime_type)
+        if isinstance(result, Exception):
+            result.document_path = document_path
+        else:
+            result['document_path'] = document_path
+        return result
+
+    def classify_documents(self, documents_paths, model_name, model_version, silent=False):
+        """
+        Submits requests for classification of multiple documents, checks the response and returns the reference ID
+        for the classified documents
+
+        :param documents_paths: Paths to the PDF files on the disk
+        :param model_name: The name of the model that was successfully deployed to be used for the classification
+        :param model_version: The version of the model that was successfully deployed to be used for the classification
+        :param silent: If set to True will not throw an exception if classification for one or more documents failed
+        :return: An iterator of objects containing the reference ID of the classified document and the classification
+        results
+        """
+        number_of_documents = len(documents_paths)
+        assert number_of_documents > 0, 'Please supply at least one document'
+        self.logger.debug(f'Started classification of {number_of_documents} documents against the model {model_name}'
+                          f' with version {model_version} in parallel using {self.polling_threads} threads')
+        with ThreadPoolExecutor(min(self.polling_threads, number_of_documents)) as pool:
+            results = pool.map(self._classify_document_wrap_errors, documents_paths, [model_name] * number_of_documents,
+                               [model_version] * number_of_documents)
+        self.logger.info(f'Finished classification of {number_of_documents} documents against the model {model_name} '
+                         f'with version {model_version}')
+        if not silent:
+            results = self._validate_results(results, 'Some documents could not be successfully classified')
+        return self._create_result_iterator(results)
```

Comment:
In case we remove the silent flag, the user has to check if (and maybe how many) Exceptions are thrown, when evaluating the generator results.

Reply:
I think this is totally fine, in previous version the user could not continue the process without the flag, the generator allows the user to ignore exceptions and continue. 
> Explicit is Better Than Implicit

Reply:
This would make the silent mode the default. I think the validation is nice for users who want to have the overview of all failures within one exception, instead of going through all the exceptions themselves.

Reply:
Silent flag removed

Reply:
Great, thanks, we will need to regenerate the API docs and the Jupiter notebooks outputs in order to get rid of the remains.